src/plugins/xmltool/stream.c

```diff
@@ -158,14 +158,14 @@ ssize_t keyToStreamBasename (const Key * key, FILE * stream, const char * parent
 		int found;
 		size_t skip = parentSize ? parentSize : elektraStrLen (parent) - 1;
 
-		found = memcmp (parent, key->key, skip);
+		found = memcmp (parent, keyName (key), skip);
```

Comment:
Is there any reason why the XML plugin used internal fields of the struct directly instead of the public API?

Reply:
The plugin is older than most parts of the public API.

Reply:
Well then it's good somebody finally got to fix it 😉